src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -150,6 +151,12 @@ public static IDisposable WriteMethodDeclaration(this CodeWriter writer, MethodS
 
         private static IDisposable WriteMethodDeclarationNoScope(this CodeWriter writer, MethodSignatureBase methodBase, params string[] disabledWarnings)
         {
+            // CSharpAtribute does not accept non-string arguments
+            if (methodBase.IsHiddenFromUser)
+            {
+                writer.Line($"[{typeof(EditorBrowsableAttribute)}({typeof(EditorBrowsableState)}.{nameof(EditorBrowsableState.Never)})]");
+            }
```

Comment:
OK if this is the reason that we add this flag, we need to change the definition of `CSharpAttribute` to make sure it could accept anything as parameters.
I believe `ValueExpression` should be a good idea. We do not really have expression when we introduce the attribute stuff

Reply:
Will create a new PR for this change.

Reply:
The main change is tied to this PR, updated here but get more change.